packages/contracts/contracts/ProofOfBurn.sol

```diff
@@ -25,29 +21,12 @@ contract ProofOfBurn {
         );
 
         semaphore.insertIdentity(_identityCommitment);
+        identityCommitments.push(_identityCommitment);
 
         emit Registered(_identityCommitment);
     }
 
-    function login(
-        bytes memory _hashPublic,
-        uint256[2] memory a,
-        uint256[2][2] memory b,
-        uint256[2] memory c,
-        uint256[4] memory input // (root, nullifiers_hash, signal_hash, external_nullifier)
-    ) public {
-        require(
-            semaphore.hasExternalNullifier(input[3]) == true,
-            "external_nullifier does not exist"
-        );
-        uint256 signalIndex = semaphore.current_signal_index();
-
-        semaphore.broadcastSignal(_hashPublic, a, b, c, input);
-
-        emit Login(signalIndex, _hashPublic, bytes32(input[3]));
-    }
-
-    function getLeaves() public view returns (uint256[] memory) {
-        return semaphore.leaves(semaphore.id_tree_index());
+    function getIdentityCommitments() public view returns (uint256[] memory) {
```

Comment:
The [`genWitness`](https://github.com/weijiekoh/libsemaphore/blob/master/ts/index.ts#L278) in libsemaphore use all identity commitment leaves to calculate the Merkle path, so here a method is added to get them.

This is a short term workaround. As the number of leaves grows, getting all the leaves on-chain would not be viable. Some off-chain techniques like getting paths from a coordinator will be developed in the near future.